ClientApp/src/app/user-details/user-details.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { Location } from '@angular/common';
+import { MatDialog } from '@angular/material/dialog';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+
+import { UserService } from '../Services/user.service';
+import { User } from '../Models/User';
+import { ResponseModel } from '../Models/ResponseModel';
+import { ConfirmActionDialogComponent } from '../confirm-action-dialog/confirm-action-dialog.component';
+import { UpdateUserModel } from '../Models/UpdateUserModel';
+
+@Component({
+  selector: 'app-user-details',
+  templateUrl: './user-details.component.html',
+  styleUrls: ['./user-details.component.css']
+})
+export class UserDetailsComponent implements OnInit {
+
+  user: User;
+  updateUser: UpdateUserModel;
+  userForm: FormGroup;
+  nameFormat = '[a-zA-Z\s]+$';
+  dateNow = new Date();
+  maxDate = new Date(new Date(this.dateNow).getTime() - 1000 * 60 * 60 * 24 * 365 * 14);
```

Comment:
Shouldn't this expression '1000 * 60 * 60 * 24 * 365 * 14' be defined as constant with some meaningful name?
Maybe even a configuration setting? (Not required but think of it).